ompi/runtime/ompi_mpi_init.c

```diff
@@ -98,7 +98,7 @@
 #endif
 #include "ompi/runtime/ompi_cr.h"
 
-#if defined(MEMORY_LINUX_PTMALLOC2) && MEMORY_LINUX_PTMALLOC2
+#if defined(MEMORY_LINUX_HAVE_MALLOC_HOOK_SUPPORT) && (MEMORY_LINUX_HAVE_MALLOC_HOOK_SUPPORT == 1)
```

Comment:
This should be replaced with
```#if MEMORY_LINUX_HAVE_MALLOC_HOOK_SUPPORT```

If this macro is not defined, then this is a configure issue